File: Models/Bid.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace UserManagement.Models
{
    public class Bid
    {
        [Key]
        public int BidID { get; set; }
        public int ProductID { get; set; }
       
        public virtual Products Products { get; set; }

        public string UserID { get; set; }
        public ApplicationUser User { get; set; }

        public double BidAmount { get; set; }
        public DateTime Timestamp { get; set; }
      
    }
}

File: Angular-FastAPI/FastAPI/app/models/bid.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from app.database import Base

class Bid(Base):
    __tablename__ = "bids"

    bid_id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    product_id = Column(Integer, ForeignKey("products.product_id", ondelete="CASCADE"), nullable=False)
    user_id = Column(String(50), ForeignKey("users.id", ondelete="NO ACTION"), nullable=False)
    user_name = Column(String(100), nullable=True)
    bid_amount = Column(Float, nullable=False)
    timestamp = Column(DateTime, default=datetime.utcnow)

    product = relationship("Product", back_populates="bids")
    user = relationship("User", back_populates="bids")
